crates/aeronet_websocket/src/server/config.rs

```diff
@@ -6,9 +6,80 @@ use {
     core::net::{Ipv6Addr, SocketAddr},
     derive_more::{Display, Error},
     rustls::pki_types::{CertificateDer, PrivateKeyDer},
-    tokio_tungstenite::tungstenite::protocol::WebSocketConfig,
+    tokio_tungstenite::tungstenite::{
+        handshake::server::{ErrorResponse, Request, Response},
+        protocol::WebSocketConfig,
+    },
 };
 
+/// Allows inspecting the handshake [`Request`] and conditionally accepting or
+/// denying it based on something like the header.
+///
+/// Useful for implementing something like netcode connect tokens to
+/// authenticate clients before allocating resources for a WS connection.
```

Comment:
```suggestion
/// Allows inspecting the handshake [`Request`] and conditionally accepting or
/// rejecting the session based on the request.
///
/// Useful for implementing connection filtering before a WebSocket connection is
/// created, for example if you want to make sure the request has a valid
/// [netcode connect token](https://github.com/mas-bandwidth/netcode/blob/main/STANDARD.md#connect-token) before accepting it.
```
make the docs a bit clearer on the use case